pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -290,21 +295,34 @@ public void isReady(@Suspended AsyncResponse asyncResponse) {
         }
     }
 
-    @GET
-    @Path("/health")
-    @ApiOperation(value = "Run a healthcheck against the broker")
-    @ApiResponses(value = {
-        @ApiResponse(code = 200, message = "Everything is OK"),
-        @ApiResponse(code = 403, message = "Don't have admin permission"),
-        @ApiResponse(code = 404, message = "Cluster doesn't exist"),
-        @ApiResponse(code = 500, message = "Internal server error")})
-    public void healthcheck(@Suspended AsyncResponse asyncResponse) throws Exception {
-        validateSuperUserAccess();
-        String heartbeatNamespace = NamespaceService.getHeartbeatNamespace(
-                pulsar().getAdvertisedAddress(), pulsar().getConfiguration());
-        String topic = String.format("persistent://%s/healthcheck", heartbeatNamespace);
+    protected void healthCheck(final AsyncResponse asyncResponse,
+                               final TopicVersion topicVersion) throws PulsarServerException {
+
+        String topic;
+        PulsarClient client;
+        try {
+            validateSuperUserAccess();
+            String heartbeatNamespace;
+
+            heartbeatNamespace = (topicVersion == TopicVersion.V1)
```

Comment:
I would make the check against v2 so the default is always the v1 format.

Reply:
here we are not doing any check based on topic name unlike topic/namespace. therefore, adding just a query param should be fine and cleaner approach.

Reply:
sorry I read @cckellogg comment incorrectly. Yes, we should check against v2

Reply:
Ok

Reply:
done